common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.ui.notifyme
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.provider.Settings
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.annotation.ChecksSdkIntAtLeast
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.startActivity
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
+import androidx.savedstate.findViewTreeSavedStateRegistryOwner
+import com.duckduckgo.mobile.android.R
+import com.duckduckgo.mobile.android.databinding.ViewNotifyMeViewBinding
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.Command
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.Command.CheckPermissions
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.Command.CheckShouldShowRequestPermissionRationale
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.Command.Close
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.Command.OpenSettings
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.Command.ShowPermissionRationale
+import com.duckduckgo.mobile.android.ui.notifyme.NotifyMeViewModel.ViewState
+import com.duckduckgo.mobile.android.ui.view.gone
+import com.duckduckgo.mobile.android.ui.view.show
+import com.duckduckgo.mobile.android.ui.viewbinding.viewBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+class NotifyMeView : FrameLayout {
```

Comment:
You can overload the constructor to avoid duplication. 

```suggestion
class NotifyMeView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = R.style.Widget_DuckDuckGo_NotifyMeView,
    ) : FrameLayout (context, attrs, defStyleAttr) {
```